src/components/MapFollowLocation/index.tsx

```diff
@@ -0,0 +1,73 @@
+import React, {useRef, useState} from 'react';
+import {FAB} from 'react-native-paper';
+import {
+  StyleURL,
+  UserLocation,
+  MapView,
+  UserTrackingMode,
+  Camera,
+} from '@rnmapbox/maps';
+import {Dimensions, StyleSheet} from 'react-native';
+import {useHeaderHeight} from '@react-navigation/elements';
+import {Portal} from 'react-native-paper';
+
+export default function MapWidget() {
+  const headerHeight = useHeaderHeight();
+  const windowHeight = Dimensions.get('window').height;
+  const [followUserMode, setFollowUserMode] = useState<UserTrackingMode | null>(
+    UserTrackingMode.FollowWithHeading,
+  );
+  const camera = useRef<Camera>(null);
+  const styles = getStyles(windowHeight - headerHeight + 12);
+
+  const followUserProps = {
+    followUserLocation: !!followUserMode,
+    ...(followUserMode && {followUserMode}),
+  };
+
+  return (
+    <Portal.Host>
```

Comment:
Why do we need to use Portal.Host here?

Reply:
It's what keeps the Fab button rendered to this component. Otherwise render in the global Portal.host and be displayed on every screen.